var/spack/repos/builtin/packages/gl2ps/package.py

```diff
@@ -31,6 +31,16 @@ class Gl2ps(CMakePackage):
     depends_on("zlib-api", when="+zlib")
     depends_on("texlive", type="build", when="+doc")
 
+    # gl2ps tries to build static and shared libs at once with the same
+    # target name. This causes ninja to fail the build
```

Comment:
Is BUILD_SHARED_LIBS only supported with the `ninja` generator?

If so, I think it would be helpful to reword this comment:

> shared builds are only supported with the ninja generator. When enabling this, the CMake files require a patch

(and I recommend putting the details about the patch in a comment right next to the patch declaration)

Reply:
It's only an issue with the Ninja generator, other generators don't care about the duplicated target caused by the import library and static library (both are `gl2ps.lib`). When the generator is anything but Ninja, you'll actually always get both the static and shared libs because the `gl2ps` CMake doesn't adhere to best practices and builds both (strongly discouraged by CMake) so it's not completely necessary to have the variant. We could just make this patch the default and force users to choose between static and shared, but I'm concerned this will break packages that rely on `gl2ps` building both static and shared. It wouldn't be the first time I've come across that scenario.

Reply:
I might still be missing some context: wouldn't building shared on Windows mean creating a `.dll` (and you could have both the `.lib` and the `.dll`?

I'm OK with changing the behavior to only build one of `shared/static` on Windows. I think more of this discussion should make it into the comment though.

Also to be clear, would this have been a problem building with `generator=ninja` on Linux? I'm wondering if

```
with when("generator=ninja"):
```

should be replaced with

```
with when("platform=windows"):
```

Reply:
Thanks, in that case I think it makes sense to make this:

```
with when("platform=windows"):
  variant(shared)
  patch(...)
```

does that sound reasonable?